### DISCUSSION TITLE
Ability to grant or deny connection based on server's certificate

### DISCUSSION THREAD

**Discussion by bostjane** [MicrosoftEdge/WebView2Feedback#2763] [Q&A]
Hi,

we're using webview2 in secure app and we miss the ability to check server's certificate and accordingly grant or deny connection. As far as I know, webview2 uses computer's certificate store to determine is server's certificate is valid. If it's not, we still have ability to grant connection via ServerCertificateErrorDetected event. But, sadly, there is no way to deny connection if server's certificate alone is OK. In other word's there should be also event like ServerCertificateOK, which should be called after webview2 determined that server's certificate is valid. In such event we would have possibility to additionally tighten/check which certificates we accept and grant or approve connection accordingly.

This would be a huge benefit if not a must in highly secure apps.

**Comment by victorthoang** [comment]
Hello @bostjane,

I'm going to create an issue based on your question. Typically if you have a question, you can create it as an Issue and we will tag it and try to answer it or assign it to a developer who can help. 

The discussion thread might be more useful for off topic questions or non-issue related inquiries!

**Reply by victorthoang** [reply]
https://github.com/MicrosoftEdge/WebView2Feedback/issues/2795
